src/SonarScanner.MSBuild.PreProcessor/JreCaching/JreCache.cs

```diff
@@ -49,41 +49,61 @@ public JreCacheResult IsJreCached(string sonarUserHome, JreDescriptor jreDescrip
 
     public async Task<JreCacheResult> DownloadJreAsync(string sonarUserHome, JreDescriptor jreDescriptor, Func<Task<Stream>> jreDownload)
     {
-        if (EnsureCacheRoot(sonarUserHome, out var cacheRootLocation)
-            && EnsureDirectoryExists(Path.Combine(cacheRootLocation, jreDescriptor.Sha256)) is { } jreDownloadPath)
+        if (!EnsureCacheRoot(sonarUserHome, out var cacheRoot)
+            || EnsureDirectoryExists(Path.Combine(cacheRoot, jreDescriptor.Sha256)) is not { } jreDownloadPath)
         {
-            var downloadTarget = Path.Combine(jreDownloadPath, jreDescriptor.Filename);
-            if (fileWrapper.Exists(downloadTarget))
-            {
-                logger.LogDebug(Resources.MSG_JreAlreadyDownloaded, downloadTarget);
-                return await UnpackJre(downloadTarget, jreDescriptor, cacheRootLocation);
-            }
-            else
-            {
-                return await DownloadAndUnpackJre(jreDownloadPath, downloadTarget, jreDescriptor, cacheRootLocation, jreDownload);
-            }
+            return new JreCacheFailure($"The JRE cache directory in '{Path.Combine(sonarUserHome, "cache", jreDescriptor.Sha256)}' could not be created.");
```

Comment:
We do call this multiple times: `Path.Combine(sonarUserHome, "cache"`. I think it deserves to be extracted to a method.

Reply:
There is also an inconsistency between the path from the if condition `Path.Combine(cacheRoot, jreDescriptor.Sha256)` and the one from the logged message `Path.Combine(sonarUserHome, "cache", jreDescriptor.Sha256)`.

I think the path should be stored in a variable and used in both places.

Reply:
`cacheRoot` might be null in this branch. Therefore I rebuild the path again from scratch.